keanu-python/build.gradle

```diff
@@ -93,6 +93,37 @@ task preparePythonEnvironment {
     dependsOn(":codegen:runCodeGeneration")
 }
 
+task formatApply {
+    dependsOn("preparePythonEnvironment")
+    doLast {
+        exec {
+            commandLine 'pipenv', 'run', 'yapf', '--style', 'google', '--in-place', '--recursive', '--parallel', '.'
+        }
+    }
+}
+
+task formatCheck {
+    dependsOn("preparePythonEnvironment")
+    doLast { 
+        exec {
+            // Exclude files that are generated
+            def keanuVetexPath = 'keanu/vertex/'
+            if(isFamily(FAMILY_WINDOWS)) {
```

Comment:
Gradle configs are just Groovy, which is a superset of Java, so you can solve this problem using Java's `Paths` class or `File.separator`

Reply:
I had tried `File.seperator` in [this](https://github.com/improbable-research/keanu/pull/349/commits/5c327b0776b2300ac939e2279f14b9b1e71ba571) commit. As shown in the Appveyor run for that commit, it also fails. It seems Windows will only exclude if the path is `.\keanu\vertex\file.py` and not `keanu\vertex\file.py`. 

Setting the path as `'.' + 'File.seperator + 'kenau' ...` won't work as yapf will error when the file starts `./` (see [here](https://github.com/google/yapf/blob/master/yapf/yapflib/file_resources.py#L117)) which would be the case for MacOS, Linux etc.

Have just tried Java's `Paths` and get the exact same issue.

Reply:
Hmm, so it's yapf's fault. I don't see why they wouldn't allow `./keanu/vertex/` as an exclude path.

Have you tried it using absolute paths? You can use the special gradle variable `$rootDir`

Reply:
The file handling doesn't seem great currently. I think ideally as Caleb mentioned we will overcome this by making sure the generated files are formatted.

I have tried absolute paths, they don't work. I think this is down to the way the matching of paths is done when it checks for the exclusions. I attempted using the absolute path for the exclusions whilst also specifying the directory to format as an absolute path. No luck with this either.